desktopexporter/app/routes/main-view.tsx

```diff
@@ -27,50 +49,100 @@ export default function MainView() {
           isFullWidth={isFullWidth}
           toggleSidebarWidth={setFullWidth.toggle}
           traceSummaries={[]}
+          numNewTraces={0}
         />
         <EmptyStateView />
       </Flex>
     );
   }
 
-  let sidebarSummaries: TraceSummaryWithUIData[] =
-    getTraceSummariesWithUIData(traceSummaries);
   return (
     <Flex height="100vh">
       <Sidebar
         isFullWidth={isFullWidth}
         toggleSidebarWidth={setFullWidth.toggle}
-        traceSummaries={sidebarSummaries}
+        traceSummaries={sidebarData.summaries}
+        numNewTraces={sidebarData.numNewTraces}
       />
       <Outlet />
     </Flex>
   );
 }
 
-function getTraceSummariesWithUIData(
+function initSidebarData(traceSummaries: TraceSummary[]): SidebarData {
+  return {
+    summaries: traceSummaries.map((traceSummary) =>
+      generateTraceSummaryWithUIData(traceSummary),
+    ),
+    numNewTraces: 0,
+  };
+}
+
+function updateSidebarData(
+  sidebarData: SidebarData,
   traceSummaries: TraceSummary[],
-): TraceSummaryWithUIData[] {
-  return traceSummaries.map((traceSummary) => {
-    if (traceSummary.hasRootSpan) {
-      let duration = getDurationNs(
-        traceSummary.rootStartTime,
-        traceSummary.rootEndTime,
-      );
-
-      let durationString = getDurationString(duration);
-      return {
-        hasRootSpan: true,
-        rootServiceName: traceSummary.rootServiceName,
-        rootName: traceSummary.rootName,
-        rootDurationString: durationString,
-        spanCount: traceSummary.spanCount,
-        traceID: traceSummary.traceID,
-      };
+): SidebarData {
+  sidebarData.numNewTraces = 0;
```

Comment:
Instead of modifying `sidebarData` in place, could we generate a new object?

```typescript
let merged : SidebarData = {
    numNewTraces: 0,
    summaries: [...sidebarData.summaries],
}
```

This would avoid needing to remember to use the spread operator above and might make the code a bit clearer

```typescript
let newSidebarData = {
    ...updateSidebarData(sidebarData, traceSummaries),
};
```